addons/phantom_camera/scripts/phantom_camera/phantom_camera_2d.gd

```diff
@@ -775,6 +778,25 @@ func update_limit_all_sides() -> void:
 			tile_map_size - Vector2(limit_margin.x, limit_margin.y) - Vector2(limit_margin.z, limit_margin.w)
 		)
 
+		# Left
+		_limit_sides.x = roundi(limit_rect.position.x)
+		# Top
+		_limit_sides.y = roundi(limit_rect.position.y)
+		# Right
+		_limit_sides.z = roundi(limit_rect.position.x + limit_rect.size.x)
+		# Bottom
+		_limit_sides.w = roundi(limit_rect.position.y + limit_rect.size.y)
+	elif _limit_node is TileMapLayer:
+		var tile_map: TileMapLayer = _limit_node as TileMapLayer
+		var tile_map_size: Vector2 = Vector2(tile_map.get_used_rect().size) * Vector2(tile_map.tile_set.tile_size) * tile_map.get_scale()
+		var tile_map_position: Vector2 = tile_map.global_position + Vector2(tile_map.get_used_rect().position) * Vector2(tile_map.tile_set.tile_size) * tile_map.get_scale()
+
+		## Calculates the Rect2 based on the Tile Map position and size + margin
+		limit_rect = Rect2(
+			tile_map_position + Vector2(limit_margin.x, limit_margin.y),
+			tile_map_size - Vector2(limit_margin.x, limit_margin.y) - Vector2(limit_margin.z, limit_margin.w)
+		)
+
```

Comment:
Direct references to the `TileMapLayer` class will not work in Godot 4.2, as the class doesn't exist in that version. So will need to use the string class checker instead — i.e. `_limit_node.is_class("TileMapLayer")`

Think since this bit, along with the other `_limit_sides` setters below, uses an identical calculation as the `TileMap` variant above, think we can consolidate it into a single section instead so we don't duplicate those lies.

So would look like:

```gdscript
if _limit_node is TileMap or _limit_node.is_class("TileMapLayer"):
    var tile_map # As we cannot typecast to TileMapLayer to support Godot 4.2 projects, we keep the type implicit until Godot 4.3 becomes the minimum supported version.

# The rest of the logic here

```

Reply:
Updated.